keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceService.kt

```diff
@@ -0,0 +1,192 @@
+import com.netflix.spinnaker.keel.events.ResourceActuationResumed
+import com.netflix.spinnaker.keel.events.ResourceActuationVetoed
+import com.netflix.spinnaker.keel.events.ResourceCheckError
+import com.netflix.spinnaker.keel.events.ResourceCheckUnresolvable
+import com.netflix.spinnaker.keel.events.ResourceCreated
+import com.netflix.spinnaker.keel.events.ResourceDeltaDetected
+import com.netflix.spinnaker.keel.events.ResourceDeltaResolved
+import com.netflix.spinnaker.keel.events.ResourceMissing
+import com.netflix.spinnaker.keel.events.ResourceTaskFailed
+import com.netflix.spinnaker.keel.events.ResourceTaskSucceeded
+import com.netflix.spinnaker.keel.events.ResourceValid
+import com.netflix.spinnaker.keel.pause.ActuationPauser
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.NoSuchDeliveryConfigException
+import com.netflix.spinnaker.keel.persistence.ResourceRepository.Companion.DEFAULT_MAX_EVENTS
+import com.netflix.spinnaker.keel.persistence.ResourceStatus
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.ACTUATING
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.CREATED
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.CURRENTLY_UNRESOLVABLE
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.DIFF
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.ERROR
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.HAPPY
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.PAUSED
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.RESUMED
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.UNHAPPY
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.UNKNOWN
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.VETOED
+import org.springframework.stereotype.Component
+
+/**
+ * Service object that offers high-level APIs for resource-related operations.
+ */
+@Component
+class ResourceService(
+  private val repository: KeelRepository,
+  private val actuationPauser: ActuationPauser
+) {
+
+  /**
+   * Returns the history of events associated with the specified resource from the database, plus any applicable
+   * application-level pause/resume events.
+   */
+  fun getEnrichedEventHistory(id: String, limit: Int = DEFAULT_MAX_EVENTS): List<PersistentEvent> {
+    val resource = repository.getResource(id)
+    return repository.resourceEventHistory(id, limit)
+      .let { events ->
+        actuationPauser.addApplicationActuationEvents(events, resource)
+      }
+  }
+
+  /**
+   * Returns the status of the specified resource by first checking whether or not it or the parent application are
+   * paused, then looking into the last few events in the resource's history.
+   */
+  fun getStatus(id: String): ResourceStatus {
+    // For the PAUSED status, we look at the `paused` table as opposed to events, since these records
```

Comment:
Calling attention to this comment. This is the only change in logic in this function and the ones just below. Otherwise, they were just moved over from `ResourceRepository`.